azure-keyvault/azure/keyvault/keys/_models.py

```diff
@@ -1,406 +1,216 @@
-     is a key backing a certificate, then managed will be true.
-    :vartype managed: bool
-    :param recovery_id: The url of the recovery object, used to identify and
-     recover the deleted key.
-    :type recovery_id: str
-    :ivar scheduled_purge_date: The time when the key is scheduled to be
-     purged, in UTC
-    :vartype scheduled_purge_date: datetime
-    :ivar deleted_date: The time when the key was deleted, in UTC
-    :vartype deleted_date: datetime
+class Key(KeyBase):
+    """A key consisting of all KeyBase and key_material information.
     """
 
-    _validation = {
-        "managed": {"readonly": True},
-        "scheduled_purge_date": {"readonly": True},
-        "deleted_date": {"readonly": True},
-    }
+    def __init__(self, attributes, key, **kwargs):
+        super(Key, self).__init__(attributes, key.kid, **kwargs)
+        self._key_material = key
+
+    @classmethod
+    def _from_key_bundle(cls, key_bundle):
+        # type: (models.KeyBundle) -> Key
+        """Construct a key from an autorest-generated KeyBundle"""
+        return cls(
+            key_bundle.attributes,
+            key_bundle.key,
+            managed=key_bundle.managed,
+            tags=key_bundle.tags,
+        )
 
-    _attribute_map = {
-        "kid": {"key": "kid", "type": "str"},
-        "attributes": {"key": "attributes", "type": "KeyAttributes"},
-        "tags": {"key": "tags", "type": "{str}"},
-        "managed": {"key": "managed", "type": "bool"},
-        "recovery_id": {"key": "recoveryId", "type": "str"},
-        "scheduled_purge_date": {"key": "scheduledPurgeDate", "type": "unix-time"},
-        "deleted_date": {"key": "deletedDate", "type": "unix-time"},
-    }
-
-    def __init__(self, **kwargs):
-        super(DeletedKeyItem, self).__init__(**kwargs)
-        self.recovery_id = kwargs.get("recovery_id", None)
-        self.scheduled_purge_date = None
-        self.deleted_date = None
+    @property
+    def key_material(self):
+        # type: () -> Dict[str, str]
+        return self._key_material
 
 
-class DeletedKeyItemPaged(Paged):
+class DeletedKey(KeyBase):
+    """A Deleted key consisting of its id, attributes, and tags, as
+    well as when it will be purged, if soft-delete is enabled for the vault.
     """
-    A paging container for iterating over a list of :class:`DeletedKeyItem <azure.keyvault.v7_0.models.DeletedKeyItem>` object
-    """
-
-    _attribute_map = {
-        "next_link": {"key": "nextLink", "type": "str"},
-        "current_page": {"key": "value", "type": "[DeletedKeyItem]"},
-    }
 
-    def __init__(self, *args, **kwargs):
-        super(DeletedKeyItemPaged, self).__init__(*args, **kwargs)
+    def __init__(self, attributes, vault_id, key_material=None, deleted_date=None, recovery_id=None, scheduled_purge_date=None, **kwargs):
+        # type: (models.KeyBase, str, Dict[str, str], Optional[datetime], Optional[str], Optional[datetime], Mapping[str, Any]) -> None
```

Comment:
the `attributes` now is of type `models.KeyBase` since we renamed it, right?
and for `key_material`, should `Optional[~azure.keyvault._generated.v7_0.models.JsonWebKeyType]` be right?